micromanager_gui/_core.py

```diff
@@ -118,26 +118,40 @@ def update_pixel_size(
         mmc.setPixelSizeConfig(resolutionID)
 
 
-def get_objective_device(obj_device: str):
-    mmc = get_core_singleton()
-    # check if there is a configuration group for the objectives
-    for cfg_groups in mmc.getAvailableConfigGroups():
-        # e.g. ('Camera', 'Channel', 'Objectives')
-
-        presets = mmc.getAvailableConfigs(cfg_groups)
-
-        if not presets:
-            continue
-
-        # first group option e.g. TINosePiece: State=1
-        cfg_data = mmc.getConfigData(cfg_groups, presets[0])
-
-        device = cfg_data.getSetting(0).getDeviceLabel()  # e.g. TINosePiece
+def get_cfg_groups_with_device_label(label: str, max_results=1) -> List[str]:
+    """Return config groups that control a device labeled `label`.
 
-        if device == obj_device:
-            STATE.objective_device = device
-            STATE.objectives_cfg = cfg_groups
-            return STATE.objective_device, STATE.objectives_cfg, presets
+    By default, returns a list of length 1 with the first matching config group.
 
-    STATE.objective_device = obj_device
-    return STATE.objective_device, None, None
+    Parameters
+    ----------
+    label : str
+        A device label
+    max_results : int, optional
+        The max number of config groups to return, by default 1
+
+    Returns
+    -------
+    List[str]
+        (up to `max_results`) config groups that contain a device named `label`
+    """
+    mmc = get_core_singleton()
+    results = []
+    _r = 0
+    groups = mmc.getAvailableConfigGroups()
+    # we use sorted here to prefer configuration groups with the same name as `label`.
+    # in the micromanager docs, `Objective` is used both as the device label and the
+    # configuration group name. So, it's worth checking that first.
+    for cfg_group in sorted(groups, key=lambda x: x != label):
+        for preset in mmc.getAvailableConfigs(cfg_group):
+            for device_label, _, _ in mmc.getConfigData(cfg_group, preset):
+                if device_label == label:
+                    results.append(cfg_group)
+                    _r += 1
+                    break
+            # only need to check the first preset, since preset share the same devices
```

Comment:
sadly this is not true. It's probably true that the mircomanager gui limits you to this, but if you edit the text of the file then you can use different devices in different presets and everything will still work. You probably shouldn't do that, but I also know that some people have done that

Reply:
I agree that that's likely safer. Though it will almost certainly break for someone out there who switches multiple things in order to switch their magnification. And it may surprise users when their config group isn't being used for that.

Reply:
Can we add docs here as well. I think I'd feel totally fine with that as long as it was well documented in an accessible way